src/hooks/__internal__/useCharacterCount/useCharacterCount.test.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from "react";
+import { render, screen, act } from "@testing-library/react";
+import useCharacterCount from ".";
+
+interface TestComponentProps {
+  value: string;
+  characterLimit?: number;
+}
+
+const TestComponent = ({ value, characterLimit }: TestComponentProps) => {
+  const [characterCount] = useCharacterCount(value, characterLimit);
+
+  return <div>{characterCount}</div>;
+};
+
+const MOCK_VALUE = "test string";
+
+test("shows characters left when input is shorter than the character limit", () => {
+  const characterLimit = 10;
+  const valueShorterThanLimit = "Test";
+  const expectedCharactersLeft = characterLimit - valueShorterThanLimit.length;
+  const expectedText = `${expectedCharactersLeft} ${
+    expectedCharactersLeft === 1 ? "character" : "characters"
+  } left`;
```

Comment:
**suggestion**: What do you think about making the expected text static? This would have the benefit of avoiding conditional logic within the test:

```suggestion
  const characterLimit = 10;
  const valueShorterThanLimit = "Test";
  const expectedText = "6 characters left";
```